### PR TITLE
feat: add service data

### PR DESCRIPTION
**Pull request by hetao92**

### CODE REVIEW

File: app/assets/components/Charts/PieChart.tsx [hunk 1]
```diff
@@ -15,56 +25,59 @@ class PieChart extends React.Component {
...
-    this.chartInstance.interaction('element-single-selected');
-    this.chartInstance.render();
+      this.chartInstance.tooltip({
+        showMarkers: false
+      });
+        
+      this.chartInstance.interval()
+        .adjust('stack')
+        .position('value')
+        .color('type', ['#063d8a', '#1770d6', '#47abfc', '#38c060'])
```
Comment (nianiaJR) [inline app/assets/components/Charts/PieChart.tsx:62]: isolate the colors config into utils?

File: app/assets/components/Charts/PieChart.tsx [hunk 2]
```diff
@@ -15,56 +25,59 @@ class PieChart extends React.Component {
...
+          }
+        });
+      this.chartInstance.render();
+    }
   }
 
   render () {
     return (
-      <div className="nebula-chart nebula-chart-pie" style={{ padding: 20 }} ref={this.chartRef} />
+      <div className="nebula-chart nebula-chart-pie" style={{ textAlign: 'center' }} ref={this.chartRef} />
```
Comment (NicolaCage) [inline app/assets/components/Charts/PieChart.tsx:80]: use calssName instead of style

File: app/assets/modules/PartitionDistribution/index.tsx [hunk 3]
```diff
@@ -1,71 +1,138 @@
...
+  return {
+    asyncGetSpaces: dispatch.nebula.asyncGetSpaces,
+    asyncUseSpaces: dispatch.nebula.asyncUseSpaces,
+    asyncGetParts: dispatch.nebula.asyncGetParts,
+  };
+};
+
+const mapState = (state: IRootState) => ({
+  spaces: state.nebula.spaces,
+  parts: state.nebula.parts,
```
Comment (NicolaCage) [inline app/assets/modules/PartitionDistribution/index.tsx:25]: where is it used?

File: app/assets/modules/Layout/routes.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import intl from 'react-intl-universal';
 
 const MachineDashboard = lazy(() => import('@assets/modules/MachineDashboard'));
 const ServiceOverview = lazy(() => import('@assets/modules/ServiceOverview'));
-const ServerMetrics = lazy(() => import('@assets/modules/ServerMetrics'));
+const QpsMetrics = lazy(() => import('@assets/modules/QpsMetrics'));
```
Comment (nianiaJR) [inline app/assets/modules/Layout/routes.tsx:6]: Qps => QPS，it's abbreviation

File: app/assets/modules/ServiceOverview/ItemCell/index.tsx [hunk 5]
```diff
@@ -3,49 +3,103 @@ import React from 'react';
...
+    this.chartInstance
+      .line()
+      .color(COLORS.LINE)
+      .position('time*value');
+    this.chartInstance
+      .area()
+      .style({
+        fill: `l(90) 0:${COLORS.AREA_TOP} 1:${COLORS.AREA_BOTTOM}`,
+      })
+      .adjust('stack')
```
Comment (nianiaJR) [inline app/assets/modules/ServiceOverview/ItemCell/index.tsx:31]: remove

File: app/assets/modules/ServiceOverview/OverviewCell/index.tsx [hunk 6]
```diff
@@ -4,19 +4,70 @@ import Icon from '@assets/components/Icon';
...
+      .line()
+      .color(COLORS.LINE)
+      .size(2)
+      .position('time*value');
+    this.chartInstance
+      .area()
+      .style({
+        fill: `l(90) 0:${COLORS.AREA_TOP} 1:${COLORS.AREA_BOTTOM}`,
+      })
+      .adjust('stack')
```
Comment (nianiaJR) [inline app/assets/modules/ServiceOverview/OverviewCell/index.tsx:33]: remove

File: app/assets/utils/interface.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
+export interface IMetric {
```
Comment (nianiaJR) [inline app/assets/utils/interface.ts:1]: interface => service-dashboard.ts?

File: app/assets/AuthorizedRoute.tsx [hunk 8]
```diff
@@ -12,6 +12,7 @@ const mapState = (state: IRootState) => ({
 const mapDispatch = () => ({});
 
 const AuthorizedRoute = ({ component: Component, render, ...rest }) => {
+  console.log('rest', rest);
```
Comment (nianiaJR) [inline app/assets/AuthorizedRoute.tsx:15]: remove？

**Review by nianiaJR — APPROVED** [review]
LGTM

**Pull request merged**